apple/internal/cc_info_support.bzl

```diff
@@ -32,7 +32,8 @@ def _get_all_deps(*, deps, split_deps_keys = []):
 
     all_deps = []
     for split_deps_key in split_deps_keys:
-        all_deps += deps[split_deps_key]
+        if split_deps_key in deps:
+            all_deps += deps[split_deps_key]
```

Comment:
@keith @brentleyjones this fixes an issue introduced by adding the `macos` attr to `apple_static_xcframework`:

```
ERROR: /Users/lpadron/Development/rules_apple/test/starlark_tests/targets_under_test/apple/BUILD:942:25: in apple_static_xcframework rule //test/starlark_tests/targets_under_test/apple:ios_static_xcframework_with_resources: 
Traceback (most recent call last):
        File "/Users/lpadron/Development/rules_apple/apple/internal/xcframework_rules.bzl", line 943, column 83, in _apple_static_xcframework_impl
                link_outputs_by_library_identifier = _group_link_outputs_by_library_identifier(
        File "/Users/lpadron/Development/rules_apple/apple/internal/xcframework_rules.bzl", line 205, column 45, in _group_link_outputs_by_library_identifier
                if swift_support.uses_swift(deps[split_attr_key]):
Error: key "ios_arm64e_device" not found in dictionary
```

How should we properly fix this? Is the `apple_rule_arm64_as_arm64e_transition` being applied here for some reason?

Reply:
This still leaves us with:

```
ERROR: /Users/lpadron/Development/rules_apple/test/starlark_tests/targets_under_test/apple/BUILD:508:8: declared output 'test/starlark_tests/targets_under_test/apple/ios_xcframework_bundling_static_fmwks.xcframework/ios-arm64_x86_64-simulator/ios_xcframework_bundling_static_fmwks.framework/Headers/ios_xcframework_bundling_static_fmwks.h' was not created by genrule. This is probably because the genrule actually didn't create this output, or because the output was a directory and the genrule was run remotely (note that only the contents of declared file outputs are copied from genrules run remotely)
```

Reply:
AFAICT adding `macos` changes the transitions to add `arm64e`, maybe this is related to #2419 as well since that's a split deps issue as well.